Makefile

```diff
@@ -76,6 +76,9 @@ run-e2e-tests:
 	# This is a test that should be run in the end to reduce the disruption to other tests because
 	# it will delete a node.
 	KUBECONFIG=${kubeconfig} go test -mod=$(MOD) -tags="$(platform),disruptive-e2e" -covermode=atomic -buildmode=exe -v ./...
+	# Test if the metrics are actually being scraped
+	KUBECONFIG=${kubeconfig} kubectl -n monitoring port-forward svc/prometheus-operator-prometheus 9090:9090 &
+	KUBECONFIG=${kubeconfig} go test -mod=$(MOD) -tags="$(platform),prometheus" -covermode=atomic -buildmode=exe -v ./...
```

Comment:
I think those tests can be just regular part of e2e testing.

Reply:
Also, when adding new tags, they should be added to the linter.

Reply:
I think these tests should be run separately because it allows us to control them when they are run.

There nature is different from the e2e tests. These can be technically categorized as e2e tests but these are being added to ensure we don't introduce regressions when upgrading prometheus operator.

Reply:
I think they are still part of e2e tests and should be run under the same tag. If developer wants to run only specific set of tests, we could allow to override the `./...`.

Reply:
I think it should be separate because it buys sometime for the prometheus to scrape metrics from the testable components.

Reply:
If we add tests for contour, openebs, metallb etc, will we have separate build tag for each? I don't think that solution scales well and using build tags has a different purpose really.

Reply:
Like what kind of tests for contour are you talking about? Why would we test contour from the perspective of non-kubernetes aspects?

Reply:
I'm talking about the scalability of given solution, not about specific examples mentioned.

Reply:
Maybe we could name it `post-e2e` then?

Reply:
I still disagree this should be named `prometheus`.